dex-it/src/test/scala/com/wavesplatform/it/sync/orders/MakerTakerFeeTestSuite.scala

```diff
@@ -0,0 +1,81 @@
+package com.wavesplatform.it.sync.orders
+
+import com.typesafe.config.{Config, ConfigFactory}
+import com.wavesplatform.dex.domain.asset.Asset.Waves
+import com.wavesplatform.dex.domain.order.OrderType.{BUY, SELL}
+import com.wavesplatform.it.MatcherSuiteBase
+
+class MakerTakerFeeTestSuite extends MatcherSuiteBase {
+
+  override protected val dexInitialSuiteConfig: Config = ConfigFactory.parseString(
+    s"""
+       |waves.dex {
+       |  price-assets = [ "$UsdId", "WAVES" ]
+       |  order-fee {
+       |    mode = dynamic
+       |    dynamic {
+       |      base-fee = 300000
+       |      zero-maker-double-taker = true
+       |    }
+       |  }
+       |}
+       """.stripMargin
+  )
+
+  override protected def beforeAll(): Unit = {
+    wavesNode1.start()
+    broadcastAndAwait(IssueUsdTx)
+    dex1.start()
+  }
+
+  private val doubledFee: Long = 2 * matcherFee
+  private val maker            = bob
+  private val taker            = alice
+
+  private def test(makerAmount: Long, takerAmount: Long, takerFee: Long, isTakerMarket: Boolean = false): Unit = {
+
+    dex1.api.cancelAll(maker)
+    dex1.api.cancelAll(taker)
```

Comment:
it seems that can be moved to afterEach method

Reply:
Done